### DISCUSSION TITLE
Why this transformation is not invertible?

### DISCUSSION THREAD

**Discussion by alfaevc** [Project-MONAI/MONAI#7818] [Q&A] [answered]
preprocessing = Compose(
        [
            LoadImaged(keys="image"),
            EnsureChannelFirstd(keys="image"),
            NormalizeIntensityd(keys="image", nonzero=True, channel_wise=True),
            ToTensor(),
            EnsureTyped(keys="image"),
        ]
    )
postprocessing = Compose(
        [
            Activationsd(keys="pred", softmax=True, dim=1),  # Apply softmax activation to predictions
            Invertd(
                 keys="pred",
                 transform=preprocessing,  # Reference to the preprocessing transform
                 orig_keys="image",
                 meta_key_postfix="meta_dict",
                 nearest_interp=False,
                 to_tensor=True
            ),
            AsDiscreted(keys="pred", argmax=True, dim=1),  # Convert predictions to discrete labels using argmax
            ScaleIntensityRanged(keys="pred", a_min=0, a_max=3, b_min=0, b_max=1),
            SaveImaged(
                keys="pred",
                meta_keys="pred_meta_dict",
                output_dir=output_path  # Replace @output_dir with actual path
            )
        ]
    )

And I got .conda/envs/mri/lib/python3.11/site-packages/monai/transforms/compose.py:355: UserWarning: inverse has been called but no invertible transforms have been supplied 
warnings.warn("inverse has been called but no invertible transforms have been supplied")

Why is this preprocessing not invertible? I've been following a similar procedure as here: https://github.com/Project-MONAI/model-zoo/blob/dev/models/wholeBrainSeg_Large_UNEST_segmentation/configs/inference.json

**Comment by KumoLiu** [comment, marked as answer]
Hi @alfaevc, thanks for your interest here.

Invertible transforms are often be useful to return them to their original space after. You might, for example, want to use a network to segment an image, and then you'll want that segmented image in the same space as the input so that you can compare them. You could then save them to file and compare them in your favourite medical image viewer.

You can double check with `isinstance(x, InvertibleTransform)`.

So in your preprocessing, all the transforms are not `InvertibleTransform` since no spacing-related transforms.

BTW, you can refer to this tutorials for more information: https://github.com/Project-MONAI/tutorials/blob/main/modules/inverse_transforms_and_test_time_augmentations.ipynb
